src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -245,12 +248,22 @@ private <T, C> void trainAndIndex(
             : NativeIndexWriter.getWriter(fieldInfo, segmentWriteState);
 
         knnVectorValues = vectorValuesRetriever.apply(vectorDataType, fieldInfo, VectorProcessingContext);
-
+        if (shouldSkipBuildingVectorDataStructure(knnVectorValues.totalLiveDocs())) {
```

Comment:
Lets add a unit test for this condition too in KNNCodecTests where if this setting is present we will validate that graph files are actually not getting created.

Reply:
Ack